mshv-ioctls/src/ioctls/vcpu.rs

```diff
@@ -0,0 +1,1160 @@
+            hv_register_name::HV_X64_REGISTER_XMM5,
+            hv_register_name::HV_X64_REGISTER_XMM6,
+            hv_register_name::HV_X64_REGISTER_XMM7,
+            hv_register_name::HV_X64_REGISTER_XMM8,
+            hv_register_name::HV_X64_REGISTER_XMM9,
+            hv_register_name::HV_X64_REGISTER_XMM10,
+            hv_register_name::HV_X64_REGISTER_XMM11,
+            hv_register_name::HV_X64_REGISTER_XMM12,
+            hv_register_name::HV_X64_REGISTER_XMM13,
+            hv_register_name::HV_X64_REGISTER_XMM14,
+            hv_register_name::HV_X64_REGISTER_XMM15,
+            hv_register_name::HV_X64_REGISTER_FP_MMX0,
+            hv_register_name::HV_X64_REGISTER_FP_MMX1,
+            hv_register_name::HV_X64_REGISTER_FP_MMX2,
+            hv_register_name::HV_X64_REGISTER_FP_MMX3,
+            hv_register_name::HV_X64_REGISTER_FP_MMX4,
+            hv_register_name::HV_X64_REGISTER_FP_MMX5,
+            hv_register_name::HV_X64_REGISTER_FP_MMX6,
+            hv_register_name::HV_X64_REGISTER_FP_MMX7,
+            hv_register_name::HV_X64_REGISTER_FP_CONTROL_STATUS,
+            hv_register_name::HV_X64_REGISTER_XMM_CONTROL_STATUS,
+        ];
+        let reg_values = self.get_reg(&reg_names)?;
+
+        let fp_control_status: hv_x64_fp_control_status_register__bindgen_ty_1 =
+            unsafe { reg_values[24].fp_control_status.__bindgen_anon_1 };
+        let xmm_control_status: hv_x64_xmm_control_status_register__bindgen_ty_1 =
+            unsafe { reg_values[25].xmm_control_status.__bindgen_anon_1 };
+        let mut ret_regs = unsafe {
+            FloatingPointUnit {
+                fpr: [[0x0; 16usize]; 8usize],
+                fcw: fp_control_status.fp_control,
+                fsw: fp_control_status.fp_status,
+                ftwx: fp_control_status.fp_tag,
+                pad1: 0x0,
+                last_opcode: fp_control_status.last_fp_op,
+                last_ip: fp_control_status.__bindgen_anon_1.last_fp_rip,
+                last_dp: xmm_control_status.__bindgen_anon_1.last_fp_rdp,
+                xmm: [[0; 16usize]; 16usize],
+                mxcsr: xmm_control_status.xmm_status_control,
+                pad2: 0x0,
+            }
+        };
+
+        for i in 0..16 {
+            unsafe {
+                ret_regs.xmm[i] =
+                    std::mem::transmute::<hv_u128, [u8; 16usize]>(reg_values[i].reg128);
+            }
+        }
+        for i in 0..8 {
+            unsafe {
+                ret_regs.fpr[i] =
+                    std::mem::transmute::<hv_u128, [u8; 16usize]>(reg_values[i].fp.as_uint128);
+            }
+        }
+
+        Ok(ret_regs)
+    }
+
+    ///
+    /// Sets a vCPU register to input value.
+    ///
+    /// # Arguments
+    ///
+    /// * `reg_name` - general purpose register name.
+    /// * `reg_value` - register value.
+    ///
+    #[cfg(not(any(target_arch = "arm", target_arch = "aarch64")))]
+    pub fn set_reg(
```

Comment:
I would move that one next to `get_reg()`, for consistency and readibility sake.